android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginViewModel.kt

```diff
@@ -73,6 +76,20 @@ class LoginViewModel(
       }
     }
 
+  private fun storeUserPreferences(application: Application, responseBody: ResponseBody) {
+    val responseBodyString = responseBody.string()
+    Timber.i(responseBodyString)
+
+    val sharedPreferences = SharedPreferencesHelper.init(application.applicationContext)
+    sharedPreferences.write(USER_SHARED_PREFERENCE_KEY, responseBodyString)
+
+    val userResponse = responseBodyString.decodeJson<UserResponse>()
+
+    if (userResponse.realmAccess.roles.contains(G6PD)) {
```

Comment:
The **realm access role**, is it something we can configure via the [ApplicationConfiguration](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/app/ApplicationConfiguration.kt) model maybe default to `null` to applications that do not need it? The `application` can be cast to [ConfigurableApplication](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/app/ConfigurableApplication.kt) to access the configuration model

Reply:
@ellykits SharedPreference default to null. Are you thinking that it would just be clearer to explicitly default it to null?

Reply:
The `null` default is ok. I was just thinking it would be good to have the realm access role as a property of `ApplicationConfiguration` so we do not need to update the logic in future for any other application that would require the role. We can save the configured value if not `null`

Reply:
@ellykits Apologies I was misunderstanding you. I think that I am with you now. You want to store the roles for future use?

The full response of `GET /userInfo` is already persisted in `SharedPreferences`
https://github.com/opensrp/fhircore/blob/quest-questionnaire-sync/android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginViewModel.kt#L83.

Do you think that it is worthwhile persisting the roles separately?

Also the roles are not really application configuration but rather user specific information so I am not sure that ApplicationConfiguration is the correct place for this data.

Let me know what you think.

Reply:
> Also the roles are not really application configuration but rather user specific information so I am not sure that ApplicationConfiguration is the correct place for this data.

I see. My only concern is how can we dynamically provide the realm access role. The current implementation has a conditional check for **G6PD**, I assume this would change for a different application.

Reply:
we can ahead with the current implementation but refactor this later, maybe introduce `UserConfiguration` that can be updated dynamically

Reply:
@ellykits Okay I think that we are finally on the same page.

Yes, this is what I was trying to say in standup.

The solution we discussed was to use the roles for this but what we actually need is a key value store.

I wanted to get this in for the deadline but I do think we need to refactor it to be more dynamic.